### PR TITLE
Added ES6 Support

### PR DESCRIPTION
**Pull request by sethmurphy18**

### CODE REVIEW

File: package.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ Package.onUse(function(api) {
   api.addFiles('event-emitter.js', 'client');
   api.addAssets('worker.js', 'client');
   api.addFiles('files.coffee', ['server', 'client']);
+  api.addAssets('exports.jsx', ['server', 'client']);
```
Comment (dr-dimitru) [inline package.js]: Are you sure it should be added as asset? Not a just file?

**Comment by sethmurphy18** [comment]
Meteor throws an error if it's added as a File...so I had to add it as an asset...all js and jsx files included are compiled and interpreted and runtime, so it shouldn't cause any major issues.

**Comment by dr-dimitru** [comment]
Hmmm... I'll play with it for a while to make sure. Because assets is directly accessible and placed into public folder.

Have you tried to add `ecmascript` package as dependency?

**Comment by sethmurphy18** [comment]
`ecmascript` is included by default on all Meteor projects now.

```
# Meteor packages used by this project, one per line.
# Check this file (and the other files in this directory) into your repository.
#
# 'meteor add' and 'meteor remove' will edit this file for you,
# but you can also edit it by hand.

meteor-base             # Packages every Meteor app needs to have
mobile-experience       # Packages for a great mobile UX
mongo                   # The database Meteor supports right now
blaze-html-templates    # Compile .html files into Meteor Blaze views
reactive-var            # Reactive variable for tracker
jquery                  # Helpful client-side library
tracker                 # Meteor's client-side reactive programming library

standard-minifier-css   # CSS minifier run for production mode
standard-minifier-js    # JS minifier run for production mode
es5-shim                # ECMAScript 5 compatibility for older browsers.
ecmascript              # Enable ECMAScript2015+ syntax in app code

autopublish             # Publish all data to the clients (for prototyping)
insecure                # Allow all DB writes from clients (for prototyping)

ostrio:files
```

If I set it to include that file as a file, meteor throws this error
```
   While building package ostrio:files:
   error: No plugin known to handle file 'exports.jsx'. If you want this file to be a static asset, use
   addAssets instead of addFiles; eg, api.addAssets('exports.jsx', 'client').
```

Worst case scenario, if someone does get that jsx file, all it has is a ES6 module definition. By default, all writes from the client are denied, so it should pose a huge issue.

**Comment by sethmurphy18** [comment]
Let me change it to a js file and see what happens

**Comment by dr-dimitru** [comment]
But, it's in your project, not in the package.

**Comment by sethmurphy18** [comment]
True, if I make it a pure js file, I get this error.

```
=> Started proxy.                                            
=> Started MongoDB.                                          
=> Errors prevented startup:                                 
                                                             
   While computing assigned variables:                       
   /packages/ostrio_files/exports.js: Unexpected token (1:30)
                                                             
   While computing assigned variables:                       
   /packages/ostrio_files/exports.js: Unexpected token (1:30)
                                                             
=> Your application has errors. Waiting for file change.     
```

**Comment by sethmurphy18** [comment]
The only other option I can think of it to rewrite the entire package in ES6 compatible lingo.

**Comment by sethmurphy18** [comment]
Actually, according to the Meteor Docs, add assets should be fine.

PackageAPI#addAssets(filenames, architecture)
Specify asset files for your package. They can be accessed via the Assets API from the server, or at the URL /packages/username_package-name/file-name from the client, depending on the architecture passed.
ARGUMENTS

filenames String or Array of Strings
Paths to the asset files.
architecture String or Array of Strings
Specify where this asset should be available (e.g., 'server', 'client', 'web.browser', 'web.cordova'). You can specify multiple architectures by passing in an array, for example ['web.cordova', 'os.linux'].

**Comment by sethmurphy18** [comment]
Oh wait, yeah, that's bad.

**Comment by dr-dimitru** [comment]
> The only other option I can think of it to rewrite the entire package in ES6 compatible lingo.

Nah... Not a solution for today

**Comment by sethmurphy18** [comment]
Exactly, that's a long process

**Comment by dr-dimitru** [comment]
Okay, I'll accept your PR, and find a way to complete it by my own, if you're okay with it, do you?

**Comment by sethmurphy18** [comment]
That's fine. I'll start working on rewriting the code as a pure ES6 module...worse case scenario, I'll create another plugin that's pure ES6.

**Comment by dr-dimitru** [comment]
Which code have you used for testing?

**Comment by sethmurphy18** [comment]
It's just a generic Meteor app. I added the package locally, and added a generic collection in the lib folder.

```javascript
import {FilesCollection} from 'meteor/ostrio:files';

this.Images = new FilesCollection({
  debug: true,
  collectionName: 'Images',
  allowClientCode: false, // Disallow remove files from Client
  onBeforeUpload: function (file) {
    // Allow upload files under 10MB, and only in png/jpg/jpeg formats
    if (file.size <= 10485760 && /png|jpg|jpeg/i.test(file.extension)) {
      return true;
    } else {
      return 'Please upload image, with size equal or less than 10MB';
    }
  }
});

if (Meteor.isServer) {
  Images.denyClient();
  Meteor.publish('files.images.all', function () {
    return Images.find().cursor;
  });

} else {

  Meteor.subscribe('files.images.all');
}
```

**Comment by dr-dimitru** [comment]
I've tried your example code on 1.6.3 (*without changes in this PR*) and it works perfectly

**Pull request merged**